src/macroexpand.scm

```diff
@@ -466,11 +466,12 @@
 
 ;; macro expander entry point
 
-(define (julia-expand-macros e)
-  (cond ((not (pair? e))     e)
+(define (julia-expand-macros-limited e max-depth)
```

Comment:
I think it might be more straightforward to define this as a boolean with a default value:
`(define (julia-expand-macros e (recursive #t))`
I'm not sure there's going to be a use case for arbitrary depths, so it's nice just to keep the implementation simple.